src/why3_tactic.ml

```diff
@@ -0,0 +1,198 @@
+  in
+  match Basics.get_args t with
+  | (Symb(s,_), [t]) when s == cfg.symb_P -> Some (translate_prop tbl t)
+  | _                                     -> None
+
+(** [translate pos builtins (hs, g)] translates from lambdapi to Why3 goal [g]
+  using the hypothesis [hs]. The function return
+  [constants, hypothesis, formula] where :
+  - [constants] maps abstracted Lambdapi terms to Why3 constants.
+  - [hypothesis] maps abstracted labels to Why3 terms (presentation of [hs]).
+  - [formula] Why3 term representing the goal [g].  *)
+let translate : Pos.popt -> sym StrMap.t -> (Env.env * term) ->
+                  cnst_table * Why3.Term.term StrMap.t * Why3.Term.term =
+    fun pos builtins (hs, g) ->
+  let cfg = get_config pos builtins in
+  let (constants, hypothesis) =
+    (* [translate_hyp (tbl, map) (name, (_, hyp))] translate the context [hyp]
+       with the label [name] and add it in [tbl] with the why3 constants. *)
+    let translate_hyp (tbl, map) (name, (_, hyp)) =
+      match translate_term cfg tbl (Bindlib.unbox hyp) with
+      | Some(tbl, why3_hyp) -> (tbl, StrMap.add name why3_hyp map)
+      | None                -> (tbl, map)
+    in
+    List.fold_left translate_hyp ([], StrMap.empty) hs
+  in
+  match translate_term cfg constants g with
+  | Some(tbl, why3_term) -> (tbl, hypothesis, why3_term)
+  | None                 ->
+      Console.fatal pos "The term [%a] is not of the form [P _]" Print.pp g
+
+(** [create tbl hyps term] adds all the constants of [tbl] in a new task, also
+    declare the hypotheses of [hyps] as axioms and register the term [term] as
+    a Why3 goal. *)
+let create : cnst_table -> Why3.Term.term StrMap.t -> Why3.Term.term ->
+               Why3.Task.task = fun tbl hyps term ->
+  (* Add the declaration of every constant. *)
+  let fn tsk (_,t) = Why3.Task.add_param_decl tsk t in
+  let tsk = List.fold_left fn None tbl in
+  (* Add the declaration of every hypothesis. *)
+  let fn name t tsk =
+    let axiom = Why3.Decl.create_prsymbol (Why3.Ident.id_fresh name) in
+    Why3.Task.add_prop_decl tsk Why3.Decl.Paxiom axiom t
+  in
+  let tsk = StrMap.fold fn hyps tsk in
+  (* Add the goal itself. *)
+  let goal = Why3.Decl.create_prsymbol (Why3.Ident.id_fresh "main_goal") in
+  Why3.Task.add_prop_decl tsk Why3.Decl.Pgoal goal term
+
+(** [prover pos prv] search and return the prover called [prv]. *)
+let prover : Pos.popt -> string -> Why3.Whyconf.config_prover = fun pos prv ->
+  (* Filter the set of why3 provers. *)
+  let fp = Why3.Whyconf.parse_filter_prover prv in
+  (* Get the set of provers. *)
+  let provers = Why3.Whyconf.filter_provers why3_config fp in
+  (* Display a message if we did not find a matching prover. *)
+  if Why3.Whyconf.Mprover.is_empty provers then
+    Console.fatal pos "[%s] not installed or not configured" prv;
+  (* Return the prover configuration. *)
+  snd (Why3.Whyconf.Mprover.max_binding provers)
+
+(** [handle prover_name ss tac ps g] FIXME *)
```

Comment:
I forgot to say that the doc is missing here.

Reply:
Added in 5e72913.